### PR TITLE
add `cupy.random.triangular`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 1]
```diff
@@ -417,6 +417,41 @@ def tomaxint(self, size=None):
...
+    def triangular(self, left, mode, right, size=None, dtype=float):
+        """Returns an array of samples drawn from the triangular distribution.
+
+        .. seealso::
+            :func:`cupy.random.triangular` for full documentation,
+            :meth:`numpy.random.RandomState.triangular`
+        """
+        left, mode, right = \
+            cupy.asarray(left), cupy.asarray(mode), cupy.asarray(right)
+        x = self.random_sample(size=size, dtype=dtype)
```
Comment (toslunar) [inline cupy/random/generator.py]: Please fix the case `size=None`.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -500,6 +500,43 @@ def tomaxint(self, size=None):
         sample &= cupy.iinfo(cupy.int_).max
         return sample
 
+    _triangular_kernel = core.ElementwiseKernel(
+        'T U, L left, M mode, R right', 'T x',
```
Comment (toslunar) [inline cupy/random/generator.py]: Why don't you overwrite `x`?
c.f. `_scale_kernel`

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: cupy/random/generator.py [hunk 3]
```diff
@@ -500,6 +500,43 @@ def tomaxint(self, size=None):
...
+            :func:`cupy.random.triangular` for full documentation,
+            :meth:`numpy.random.RandomState.triangular`
+        """
+        left, mode, right = \
+            cupy.asarray(left), cupy.asarray(mode), cupy.asarray(right)
+        if size is None:
+            size = cupy.broadcast(left, mode, right).shape
+        x = self.random_sample(size=size, dtype=dtype)
+        RandomState._triangular_kernel(left, mode, right, x)
+        return x
```
Comment (toslunar) [inline cupy/random/generator.py]: Could you return the output of the kernel?

File: cupy/random/generator.py [hunk 4]
```diff
@@ -500,6 +500,43 @@ def tomaxint(self, size=None):
...
+
+    def triangular(self, left, mode, right, size=None, dtype=float):
+        """Returns an array of samples drawn from the triangular distribution.
+
+        .. seealso::
+            :func:`cupy.random.triangular` for full documentation,
+            :meth:`numpy.random.RandomState.triangular`
+        """
+        left, mode, right = \
+            cupy.asarray(left), cupy.asarray(mode), cupy.asarray(right)
```
Comment (toslunar) [inline cupy/random/generator.py]: Could you check `left <= mode`, `mode <= right`, and `left < right`?

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**